misc/python/materialize/mzcompose/services.py

```diff
@@ -39,12 +39,18 @@ def __init__(
     ) -> None:
         if environment is None:
             environment = [
-                "MZ_LOG_FILTER",
                 "MZ_SOFT_ASSERTIONS=1",
+                "MZ_METRICS_SCRAPING_INTERVAL=1s",
+                # Please think twice before forwarding additional environment
+                # variables from the host, as it's easy to write tests that are
+                # then accidentally dependent on the state of the host machine.
+                #
+                # To dynamically change the environment during a workflow run,
+                # use Composition.override.
+                "MZ_LOG_FILTER",
                 "AWS_ACCESS_KEY_ID",
                 "AWS_SECRET_ACCESS_KEY",
                 "AWS_SESSION_TOKEN",
-                "MZ_METRICS_SCRAPING_INTERVAL=1s",
```

Comment:
This is needed here so that when a test runs a version that understands `MZ_METRICS_SCRAPING_INTERVAL`, the option is put into effect and used (but if the upgrade test runs an older Mz version that does not have this option, nothing bad happens).

Otherwise I believe a default of 30s was kicking in and it was detrimental for some main-dir testdrive tests. lets see if CI is green, if it is, good riddance.

Reply:
Oh no! I meant to just reorder these, not lose some of them. 🙈 Thanks for the eagle eyes. I'm not sure I follow though: is it okay to merge with this removed, or do you want me to add it back for the upgrade test?

Reply:
Oh wait, I think it _is_ still there up top! I'm not crazy, am I?

Reply:
I've decided I'm not crazy. Merging to unblock some further conflicting changes.